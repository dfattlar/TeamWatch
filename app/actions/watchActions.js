"use strict";

import * as types from "./actionTypes";

export function startWatch(intervalId) {
  return {
    type: types.START_WATCH,
    startTime: Date.now(),
    intervalId: intervalId
  };
}

export function stopWatch() {
  return {
    type: types.STOP_WATCH
  };
}

export function resetAll() {
  return {
    type: types.RESET_ALL
  };
}

export function resetTime() {
  return {
    type: types.RESET_TIME
  };
}

export function resetAthleteList() {
  return {
    type: types.RESET_ATHLETE_LIST
  };
}

export function tick() {
  return {
    type: types.TICK,
    time: Date.now()
  };
}

export function openModal() {
  return {
    type: types.OPEN_MODAL
  };
}

export function closeModal() {
  return {
    type: types.CLOSE_MODAL
  };
}

export function newAthleteInputFirst(firstName) {
  return {
    type: types.NEW_ATHLETE_INPUT_FIRST,
    newAthleteInputFirst: firstName
  };
}

export function newAthleteInputLast(lastName) {
  return {
    type: types.NEW_ATHLETE_INPUT_LAST,
    newAthleteInputLast: lastName
  };
}

export function addAthlete() {
  return {
    type: types.ADD_ATHLETE
  };
}

export function addAthleteError() {
  return {
    type: types.ADD_ATHLETE_ERROR
  };
}

export function addSplit(id) {
  return {
    type: types.ADD_SPLIT,
    id: id,
    splitTime: Date.now()
  };
}

export function modeChange(mode) {
  return {
    type: types.MODE_CHANGE,
    timerMode: mode
  };
}

export function addHistory(watchData) {
  return {
    type: types.ADD_HISTORY,
    payload: watchData
  };
}
