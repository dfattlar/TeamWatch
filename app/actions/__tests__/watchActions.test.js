import * as actions from '../watchActions';

describe('Watch Actions', () => {
  it('should create an action to start the watch', () => {
    Date.now = jest.fn().mockReturnValue(10);
    expect(actions.startWatch()).toMatchSnapshot();
    jest.resetAllMocks();
  });

  it('should create an action to stop the watch', () => {
    expect(actions.stopWatch()).toMatchSnapshot();
  });

  it('should create an action to reset all values on watch', () => {
    expect(actions.resetAll()).toMatchSnapshot();
  });

  it('should create an action to reset time values on watch', () => {
    expect(actions.resetTime()).toMatchSnapshot();
  });

  it('should create an action to reset time values on watch', () => {
    expect(actions.resetAthleteList()).toMatchSnapshot();
  });

  it('should create an action to open the modal', () => {
    expect(actions.closeModal()).toMatchSnapshot();
  });

  it('should create an action to add a split', () => {
    Date.now = jest.fn().mockReturnValue(10);
    const id = 'id_123';
    expect(actions.addSplit(id)).toMatchSnapshot();
    jest.resetAllMocks();
  });

  it('should create an action to change the mode (race,relay)', () => {
    const timerMode = 'MODE';
    expect(actions.modeChange(timerMode)).toMatchSnapshot();
  });

  it('should create an action to add history', () => {
    const watchData = {time: 0};
    expect(actions.addHistory(watchData)).toMatchSnapshot();
  });
});
