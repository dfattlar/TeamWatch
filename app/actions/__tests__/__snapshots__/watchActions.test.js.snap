// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch Actions should create an action to add a split 1`] = `
Object {
  "id": "id_123",
  "splitTime": 10,
  "type": "ADD_SPLIT",
}
`;

exports[`Watch Actions should create an action to add history 1`] = `
Object {
  "payload": Object {
    "time": 0,
  },
  "type": "ADD_HISTORY",
}
`;

exports[`Watch Actions should create an action to change the mode (race,relay) 1`] = `
Object {
  "timerMode": "MODE",
  "type": "MODE_CHANGE",
}
`;

exports[`Watch Actions should create an action to open the modal 1`] = `
Object {
  "type": "CLOSE_MODAL",
}
`;

exports[`Watch Actions should create an action to reset all values on watch 1`] = `
Object {
  "type": "RESET_ALL",
}
`;

exports[`Watch Actions should create an action to reset time values on watch 1`] = `
Object {
  "type": "RESET_TIME",
}
`;

exports[`Watch Actions should create an action to reset time values on watch 2`] = `
Object {
  "type": "RESET_ATHLETE_LIST",
}
`;

exports[`Watch Actions should create an action to start the watch 1`] = `
Object {
  "startTime": 10,
  "type": "START_WATCH",
}
`;

exports[`Watch Actions should create an action to stop the watch 1`] = `
Object {
  "stopTime": undefined,
  "type": "STOP_WATCH",
}
`;
